002_simulation_structure/002_simulation_structure.tex

```diff
@@ -1,23 +1,109 @@
+
+\subsection{Ticks \& Timing Details}
+
+Discrete time in the simulation is represented by ticks, the smallest unit of time. During each tick all agents can perform an action such as eat or send a message. This means that the amount of ticks per day restricts the communication between the agents. Moreover, given that messages can only be passed from one floor to one below or above on each tick, we can conclude that certain messages will never get to their receiver. The amount of ticks per day is set in the initialisation of the simulation and it is the product of the ticks per floor and the number of floors. This method was preferred to having a constant amount of ticks a day because that would cause problems with division with the number of floors (decimal numbers).
+
+\subsection{Agent Configuration}
+
+Agent structures are separated into two different aspects; a base agent and a custom agent – both of which are required to represent a single agent.
+The base agent component contains the core information of an agent, including its Health Points, Floor and UUID\footnote{UUID - Universally Unique Identifier}, as well as pointer to the tower structure. 
+The custom agent contains a pointer to its own base agent and any parameters relevant to the specific agent behaviour, this varies between agent types and was implemented by agent teams. To access information the custom agent calls 'getter' functions in its base agent.
+
+The core parameters of an agent were abstracted out of the custom agent for two reasons.
+\begin{enumerate}
+    \item Custom agents cannot manipulate their own data or information, rather they can only request access through the mentioned 'getter' functions, which either returns information or an error.
+    \item The tower struct has access to the base agents and can therefore influence the agent parameters as determined by the laws existing within the tower. Examples of this include floor reshuffling and health decay.
+\end{enumerate}
+
+\subsection{Death \& Replacement}
+
+Deaths occur in the Tower when an Agent has not eaten enough food to stay alive over a certain duration. The nature of replacement in the tower could be of several types:
+
+\begin{enumerate}
+    \item Replacing Agents with Random Agent Type
+    \item Replacing Agents with the same Agent Type
+    \item Replacing Agents with the Dominant Agent Type
+\end{enumerate}
+
+Our design chose to replace agents with the same agent type. While replacing agents with a random agent type would introduce an element of uncertainty, there made it difficult to analyse individual agent behaviours. Replacing agents with the dominant agent type would have been an interesting experimentation to run. However, this posed the question of defining what the most “dominant” agent would be. would not be equal amounts of all agents in the tower at a given time.
```

Comment:
```suggestion
Our design chose to replace agents with the same agent type. While replacing agents with a random agent type would introduce an element of uncertainty, there made it difficult to analyse individual agent behaviours. It would have been interesting to experiment with replacing agents with the dominant agent type, however this posed the question of defining what the most “dominant” agent would be. would not be equal amounts of all agents in the tower at a given time.
```

Reply:
Also not sure what's going on with the last sentence